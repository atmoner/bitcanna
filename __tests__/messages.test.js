import { BitCannaGlobal } from '../src/proto';

it('messages', async () => {
  expect(
    BitCannaGlobal.bcna.bcna.MessageComposer.withTypeUrl.createBitcannaid({
      address: 'osmo1f4vxvvvvvvvvvv3luuddddddddddcccccccccc',
      bcnaid: 'myid',
      creator: 'creator'
    })
  ).toMatchSnapshot();
});

it('queries', async () => {
  const client = new BitCannaGlobal.bcna.bcna.QueryClient({
    restEndpoint: 'https://bitcanna'
  });
  client.request = async (url) => {
    // modify for testing
    return url;
  };
  const res = await client.bitcannaid({ id: 'someid' });
  console.log(res);
});
