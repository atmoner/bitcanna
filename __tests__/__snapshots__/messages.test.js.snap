// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages 1`] = `
Object {
  "typeUrl": "/BitCannaGlobal.bcna.bcna.MsgCreateBitcannaid",
  "value": Object {
    "address": "osmo1f4vxvvvvvvvvvv3luuddddddddddcccccccccc",
    "bcnaid": "myid",
    "creator": "creator",
  },
}
`;
